### PR TITLE
Tool button component refactored to also use as toggle button

### PR DESCRIPTION
**Pull request by ruchimotwaniBC**

### CODE REVIEW

File: src/components/ToolButton.tsx [hunk 1]
```diff
@@ -33,6 +33,8 @@ interface ToolButtonProps {
   disabled?: boolean;
   tooltipText?: string;
   className?: string;
+  toggle?: boolean;
+  customSx?: boolean;
```
Comment (forman) [inline src/components/ToolButton.tsx:37]: I'd prefer if we just name it `sx`:

```suggestion
  sx?: boolean;
```

File: src/components/ToolButton.tsx [hunk 2]
```diff
@@ -41,20 +43,39 @@ const ToolButton: React.FC<ToolButtonProps> = ({
...
       size="small"
+      sx={{
+        ...(toggle && {
+          border: "1px solid rgba(0, 0, 0, 0.12)",
+          borderRadius: "4px",
+          padding: "2.4px",
+          color: isToggled ? "rgba(0, 0, 0, 0.87)" : "rgba(0, 0, 0, 0.54)",
+          backgroundColor: isToggled ? "rgba(0, 0, 0, 0.08)" : "transparent",
+        }),
+      }}
```
Comment (forman) [inline src/components/ToolButton.tsx:78]: Use MUI's `ToggleButton`  <https://mui.com/material-ui/react-toggle-button/>.
Why didn't you use it?

File: src/components/ToolButton.tsx [hunk 3]
```diff
@@ -41,20 +43,39 @@ const ToolButton: React.FC<ToolButtonProps> = ({
...
   tooltipText,
+  toggle = false,
 }) => {
+  const [isToggled, setIsToggled] = useState(false);
+  const handleClick = (event: MouseEvent<HTMLButtonElement>) => {
+    if (toggle) {
+      setIsToggled(!isToggled);
+    }
+    onClick(event);
+  };
```
Comment (forman) [inline src/components/ToolButton.tsx:54]: Why should the toggled state change? Toggle buttons change between selected and unselected state, but they should not change the toggled state.

**Review by forman — CHANGES_REQUESTED** [review]

File: src/components/ToolButton.tsx (hunk 1 above)
Comment (ruchimotwaniBC) [reply src/components/ToolButton.tsx:37]: Sorry  its mistake I forgot to remove it, not using it anymore. Will just remove it

**Review by forman — APPROVED** [review]

**Pull request merged**